docs/mp/guides/upgrade_4x.adoc

```diff
@@ -0,0 +1,191 @@
+
+* *MicroProfile OpenAPI {version-lib-microprofile-openapi-api}*:
++
+Incompatible changes described in link:https://download.eclipse.org/microprofile/microprofile-open-api-3.1/microprofile-openapi-spec-3.1.html#incompatible_changes_30[MicroProfile OpenAPI {version-lib-microprofile-openapi-api} Specification]
+
+* *MicroProfile Rest Client {version-lib-microprofile-rest-client}*:
++
+Incompatible changes described in link:https://download.eclipse.org/microprofile/microprofile-rest-client-3.0/microprofile-rest-client-spec-3.0.html#_incompatible_changes[MicroProfile Rest Client {version-lib-microprofile-rest-client} Specification]
+
+=== Supported Jakarta EE specifications
+
+* *CDI (Jakarta Contexts and Dependency Injection) {version-lib-jakarta-cdi}*:
++
+Changes described in link:https://jakarta.ee/specifications/cdi/4.0/jakarta-cdi-spec-4.0.html#architecture[CDI (Jakarta Contexts and Dependency Injection) {version-lib-jakarta-cdi} Specification]
+
+* *JAX-RS (Jakarta RESTful Web Services) {version-lib-jakarta-jaxrs-api}*:
++
+Changes described in link:https://download.eclipse.org/microprofile/microprofile-rest-client-3.0/microprofile-rest-client-spec-3.0.html#_incompatible_changes[JAX-RS (Jakarta RESTful Web Services) {version-lib-jakarta-jaxrs-api}Specification]
+
+* *JSON-B (Jakarta JSON Binding) {version-lib-jakarta-jsonb-api}*:
++
+Changes described in link:https://jakarta.ee/specifications/jsonb/2.0/jakarta-jsonb-spec-2.0.html#change-log[JSON-B (Jakarta JSON Binding) {version-lib-jakarta-jsonb-api} Specification]
+
+
+* *Jakarta Annotations {version-lib-jakarta-annotations-api}*:
++
+Full information in link:https://jakarta.ee/specifications/annotations/2.0/annotations-spec-2.0.html[Jakarta Annotations {version-lib-jakarta-annotations-api} Specification]
+
+* *Jakarta Persistence API {version-lib-jakarta-persistence-api}*:
++
+Changes described in link:https://jakarta.ee/specifications/persistence/3.0/jakarta-persistence-spec-3.0.html#revision-history[Jakarta Persistence API {version-lib-jakarta-persistence-api} Specification]
+
+* *Jakarta Transactions API {version-lib-jakarta-transaction-api}*:
++
+Changes described in link:https://jakarta.ee/specifications/transactions/2.0/jakarta-transactions-spec-2.0.html#revision-history[Jakarta Transactions API {version-lib-jakarta-transaction-api} Specification]
+
+* *Jakarta WebSocket API {version-lib-jakarta-websockets-api}*:
++
+Changes described in link:https://jakarta.ee/specifications/websocket/2.0/websocket-spec-2.0.html#changes[Jakarta WebSocket API {version-lib-jakarta-websockets-api} Specification]
+
+* *Jakarta Bean Validation {version-lib-jakarta-bean-validation}*:
++
+Changes described in link:https://jakarta.ee/specifications/bean-validation/2.0/bean-validation_2.0.html#changelog[Jakarta Bean Validation {version-lib-jakarta-bean-validation} Specification]
+
+NOTE: Please, read each specification carefully for incompatible changes!
+
+=== Significant changes
+
```

Comment:
Is this where a new Security section would be added and David's info could be placed?